packages/language-support/src/autocompletion/completionCoreCompletions.ts

```diff
@@ -41,6 +41,32 @@ function backtickIfNeeded(e: string): string | undefined {
   }
 }
 
+function backtickDbNameIfNeeded(e: string): string | undefined {
+  if (e == null || e == '') {
+    return undefined;
+  } else if (/[^\p{L}\p{N}_.]/u.test(e) || /[^\p{L}_]/u.test(e[0])) {
+    return `\`${e}\``;
+  } else {
+    return undefined;
+  }
+}
+
+function backtickParamIfNeeded(e: string): string | undefined {
+  if (e == null || e == '') {
+    return undefined;
+  }
+  const namePart = e.slice(1);
+  if (e[0] == '$') {
+    if (/[^\p{L}\p{N}_]/u.test(namePart) || /[^\p{L}_]/u.test(namePart[0])) {
+      return '$' + `\`${namePart}\``;
+    } else {
+      return undefined;
+    }
+  } else {
+    return backtickIfNeeded(e);
+  }
+}
```

Comment:
You could refactor all this method to depend on the `backtickIfNeeded` as:

```ts
function backtickParamIfNeeded(e: string): string | undefined {
  if (e == null || e == '') {
    return undefined;
  }
  const namePart = e.slice(1);
  const backticked = backtickIfNeeded(namePart);
  if (backticked) {
    return '$' + backticked;
  } else {
    return undefined;
  }
}

Reply:
Looks like fixing comment 2 will actually mean I can get ahead of adding the $ and just use backtickIfNeeded from scratch